status.js

```diff
@@ -21,33 +21,61 @@
 // THE SOFTWARE.
 'use strict';
 
-var createTable = require('./lib/table.js');
-var ClusterManager = require('./lib/cluster.js');
+var assertTruthy = require('./lib/util.js').assertTruthy;
+var Cluster = require('./lib/cluster.js');
 var parseStatusCommand = require('./parser.js').parseStatusCommand;
+var printTable = require('./lib/table.js').print;
+
+function getDampScoreRange(allStats, statusMember) {
+    var lowest = Number.MAX_VALUE;
+    var highest = 0;
+
+    allStats.forEach(function each(stat) {
+        stat.members.forEach(function each(member) {
+            if (member.address !== statusMember.address) return;
+
+            var dampScore = member.dampScore;
+            if (typeof dampScore === 'undefined') {
+                return;
+            }
+
+            if (dampScore < lowest) {
+                lowest = dampScore;
+            }
+
+            if (dampScore > highest) {
+                highest = dampScore;
+            }
+        });
+    });
+
+    return lowest + '..' + highest;
+}
 
 function main() {
     var command = parseStatusCommand();
-    var clusterManager = new ClusterManager({
+    var cluster = new Cluster({
         useTChannelV1: command.useTChannelV1,
         coordAddr: command.coordinator
     });
-    clusterManager.fetchStats(function onStats(err) {
-        if (err) {
-            console.error('Error: ' + err.message);
-            process.exit(1);
-        }
-
-        if (clusterManager.getPartitionCount() > 1) {
-            console.error('Error: cluster is partitioned. An accurate status cannot be provided.');
-            process.exit(1);
-        }
-
-        var table = createTable([]);
-        var cluster = clusterManager.getClusterAt(0);
-        cluster.membership.forEach(function each(member) {
-            table.push([member.address, member.status]);
+    cluster.fetchStats(function onStats(err) {
+        assertTruthy(!err, (err && err.message));
+        assertTruthy(cluster.getPartitionCount() === 1,
```

Comment:
A discussion outside the scope of this PR, but we should talk about whether the `status` command can tell us something useful even when a ring is partitioned. I previously found it weird that the `status` command only gives the status if the ring is healthy. Part of the reason you might want the status is if you know your ring is unhealthy and you're trying to get more information.